changes/views.py

```diff
@@ -210,3 +210,34 @@ def search_optouts_message_type():
         'metric_name': total_key,
         'metric_value': total,
     })
+
+
+class ReceiveAdminOptout(mixins.CreateModelMixin,
+                         generics.GenericAPIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, *args, **kwargs):
+
+        try:
+            data = utils.json_decode(request.body)
+        except ValueError as e:
+            return JsonResponse({'reason': "JSON decode error",
+                                'details': six.text_type(e)}, status=400)
+
+        try:
+            identity_id = data['identity']
+        except KeyError as e:
+            return JsonResponse({'reason': '"identity" must be specified.'},
+                                status=400)
```

Comment:
Instead of doing all of this validation ourselves, we can just use a serializer: http://www.django-rest-framework.org/api-guide/serializers/

All that we really need to do, is add the `action`, `data`, and `source` fields here, the validation and checking of the request and the `mother_id` field can be done in the serializer.